contributors/THIRD-PARTY-EXTENSION.md

```diff
@@ -0,0 +1,119 @@
+## Introduction
+
+Welcome to the guide for developing third party extensions in the `create-eth` repository.
+
+Third party extension allows developers to **extend** base instance created by `npx create-eth@latest` and publish the extension to github that can be used by other developers via:
+
+```bash
+npx create-eth@latest -e {your-github-userName}/{extension-repo-name}:{extension-branch-name} # extension-branch-name is optional
+```
+
+It is important to understand distinction between instance and extension.
+
+- Instance is created by `npx create-eth@latest` and contains full project files.
+- Extension does not contain full project files, instead it only contains extra files, extra directories and [`*.args.mjs`](TEMPLATING.md#args-files) files that will added to the base instance created via `npx create-eth@latest` with `-e` flag.
+
+> TIP: Checkout directory structure of `eip-712` extension maintained Scaffold-ETH team at https://github.com/scaffold-eth/create-eth-extensions/tree/eip-712
+
+This document outlines the workflow for creating and testing your extensions.
+
+## Workflow Overview
+
+An developer goes through 2 phases while developing extension:
+
+- **Phase 1**: Initial Setup and Development
+  - This phase helps you create your own extension based on the changes you make to instance created by `npx create-eth@latest`
+- **Phase 2**: Local Testing and Publishing
+  - This phase helps you to try out your extension locally and see how it works when used by other developers before publishing to github
+
+### Phase 1: Initial Setup and Development
+
+1. **Clone the `create-eth` Repository:**
+
+   ```bash
+   git clone https://github.com/scaffold-eth/create-eth.git
+   cd create-eth
```

Comment:
May want to consider adding `yarn install` here so dev is fully ready to run the script alias that comes next
```suggestion
   git clone https://github.com/scaffold-eth/create-eth.git
   cd create-eth
   yarn install
```